apps/web/playwright/booking/phoneQuestione2e/phoneQuestion.e2e.ts

```diff
@@ -0,0 +1,240 @@
+import { loginUser } from "../../fixtures/regularBookings";
+import { test } from "../../lib/fixtures";
+import { setupEventTypeAndExecuteBooking } from "../utils/bookingUtils";
+
+test.describe("Booking With Phone Question and Each Other Question", () => {
+  test.beforeEach(async ({ page, users }) => {
+    await loginUser(users);
+    await page.goto("/event-types");
+  });
+
+  test.describe("Booking With Phone Question and Address Question", () => {
+    const bookingOptions = { hasPlaceholder: true, isRequired: true };
+    test("Phone and Address required", async ({ page }) => {
+      await setupEventTypeAndExecuteBooking(
```

Comment:
Alright, that makes sense, we will provide these changes 😄

Reply:
Hi @ThyMinimalDev We are finalizing the changes but we need some clarification, what do you think about the fillAndConfirmBooking function? Should we remove the function and write the code explicitly for each case using the bookingPage fixture as well?
Take a look at this picture:

![image](https://github.com/calcom/cal.com/assets/121884634/6440c7a7-b573-42e5-9367-d19e4f8d0987)

What do you think about that?

Reply:
Hello there, great work, one thing I would recommend, when having functions with lots of parameters (more than 3 I would say), is to convert to a single json param, for example fillAndConfirmBooking would be called like that:

```
await bookpingPage.fillAndConfirmBooking({
page: eventTypePage,
placeholder: "Please share anything...",
question: "phone",
...
})
```

the goal of this is again to improve readability and avoid having to remember the prototype of the function to know which parameter does what.

otherwise, looking at the content of fillAndConfirmBooking  the function, I think it's short and clear enough to not need splitting.